### PR TITLE
Fix for WG rlogin extra chars

### PR DESCRIPTION
**Pull request by synacktic**
Simple fix for the Worldgroup rlogin. Not much else to say really.

### CODE REVIEW

**Comment by enusbaum** [comment]
I don't think this extra data is part of the rlogin payload. I'm debugging at that point and here's the state during calling lonrou after connecting via rlogin from wg3nt:

image

**Comment by enusbaum** [comment]
I can confirm the data comes in later after `lonrou` has been called.

Modifying `ProcessIncomingClientData` in `RloginSession.cs` with this code seems to work:

```csharp

 if (SessionState != EnumSessionState.Negotiating)
            {
                //Ugly WG3NT RLOGIN Extra Data Hack
                if (bytesReceived == 12 && clientData[0] == 255 && clientData[1] == 255)
                    return (null, 0);

                return (clientData, bytesReceived);
            }

```

It identifies a signature of the incoming weird data from Worldgroup's Rlogin client and drops it. 

A little hack-y, but it works? 🤷‍♂️

**Review by enusbaum — CHANGES_REQUESTED** [review]
See comment on main PR thread

**Comment by synacktic** [comment]
It was working on my side, so I'm guessing it just had to do with processor speed in that case. I'll look over the dirty hack, and use it or something like it. Would you like me to remove or leave the .clear() extension @enusbaum ?

**Comment by synacktic** [comment]
Ok, now I think we wanted to add a config item, maybe "Rlogin.Behavior" or "Rlogin.Compat" or "Rlogin.Source" or some such to specify certain behaviors based upon the client that will be connecting.

One concern is, might someone want to connect from more than one type of system and need to be able to select certain behaviors?

I know which bit is a fix for WGNT, but I don't know what code might be doing work around for mystic and synchronet...

Mystic:  #76  ?
WG #425 
Synchronet ??

TLDR; What do we want this config setting to look like and what code will it enable / disable?

**Comment by enusbaum** [comment]
Let's call it `Rloing.Compatibility` and we'll define an enumerator for `WG3NT`, and `None` (or `Default`)

We'll use the proposed hack but wrap it to only invoke if the `WG3NT` option is selected for compatibility. If the option is `None` or missing entirely, the compatibility code won't be invoked.

**Comment by synacktic** [comment]
I'm sure I went about this in a funny way, but you can always course correct me.

File: MBBSEmu/Configuration.cs [hunk 1]
```diff
@@ -64,6 +64,17 @@ public AppSettings(ILogger logger)
...
         public string RloginIPAddress => GetIPAppSettings("Rlogin.IP");
         
+        public Session.Rlogin.EnumRloginCompatibility RloginCompatibility
+        {
+            get
+            {
+                if (!ConfigurationRoot.GetSection("Rlogin.Compatibility").Exists())
+                    return Session.Rlogin.EnumRloginCompatibility.Default;
+                    
+                return (Session.Rlogin.EnumRloginCompatibility)Enum.Parse(typeof(Session.Rlogin.EnumRloginCompatibility), ConfigurationRoot["Rlogin.Compatibility"]);
```
Comment (enusbaum) [inline MBBSEmu/Configuration.cs:74]: Change this to an `Enum.TryParse` to be safer if they type in a value that can't be matched

**Review by enusbaum — CHANGES_REQUESTED** [review]
Just the one change, looks good!

File: MBBSEmu/Configuration.cs (hunk 1 above)
Comment (synacktic) [reply MBBSEmu/Configuration.cs]: It is not a drop in replacement, so I changed things around a bit.

**Review by enusbaum — APPROVED** [review]
